### PR TITLE
Solution: when adding and editing the monitoring body as x-www-form-urlencoded, enter the key in the body and then enter the value, and the parameter position will change

### PR DESCRIPTION
**Pull request by wuhuiluo**
## What this PR does / why we need it:
Solution: when adding and editing the monitoring body as x-www-form-urlencoded, enter the key in the body and then enter the value, and the parameter position will change

## Does this PR introduce a user interface change?

❎ No

## ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |              |
| 🇨🇳 中文    |              |

## Does this PR need be patched to older version?
✅ Yes(version is required)
release/1.4

## Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: shell/app/common/components/key-value-table/index.tsx [hunk 1]
```diff
@@ -223,7 +223,7 @@ class KeyValueTable extends React.Component<IProps, IState> {
     const { onChange } = this.props;
     (find(dataSource, { uniKey }) as object)[name] = value;
     this.setState({ dataSource });
-    onChange && onChange(convertToMapData(dataSource));
+    onChange && onChange(Array.isArray(this.props.data) ? dataSource : convertToMapData(dataSource));
```
Comment (daskyrk) [inline shell/app/common/components/key-value-table/index.tsx:226]: type not update

Comment (wuhuiluo) [reply shell/app/common/components/key-value-table/index.tsx:226]: ok,modified

**Pull request merged**